### PR TITLE
[jit][fuser] Remove hard-coded NVRTC specific constant from fuser header

### PR DESCRIPTION
**Pull request by bwasti**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22699 [jit][fuser] Remove hard-coded NVRTC specific constant from fuser header**

Differential Revision: [D16192290](https://our.internmc.facebook.com/intern/diff/D16192290)

### CODE REVIEW

File: torch/csrc/jit/passes/graph_fuser.cpp [hunk 1]
```diff
@@ -138,6 +138,9 @@ struct GraphFuser {
   FusionCallback callback_ = [&](Node* n) { return isFusableDefault(n); };
   Symbol kind_ = prim::FusionGroup;
 
+  // Default limit works for nvrtc limitations, change with setInputArgLimit
+  size_t subgraph_arg_limit_ = 128;
```
Comment (yinghai) [inline torch/csrc/jit/passes/graph_fuser.cpp]: Maybe we should still copy all the comments.

**Review by yinghai — APPROVED** [review]

File: torch/csrc/jit/fuser/compiler.cpp [hunk 2]
```diff
@@ -332,10 +332,6 @@ std::shared_ptr<FusedKernel> compileKernel(
     }
   }
 
-  // Have checked the limit at graph_fuser. Assert nothing else changing that.
-  AT_ASSERT((flat_inputs.size() + flat_outputs.size()) <=
```
Comment (wanchaol) [inline torch/csrc/jit/fuser/compiler.cpp]: Can you elaborate more about the motivation? with this change fusion compiler could not catch the case where it reach over the arguments limit (although it's rare).

Comment (zrphercule) [reply torch/csrc/jit/fuser/compiler.cpp]: we would like to extend the arguments limit here. Not sure if this assertion should be removed.

Comment (wanchaol) [reply torch/csrc/jit/fuser/compiler.cpp]: Then I think we can still leave the `fusion_kernel_args_limit = 128` in the header with no const expr, and set the limit in graph fuser? that way the fusion compiler can still catch the overflow on argument size.

Comment (bwasti) [reply torch/csrc/jit/fuser/compiler.cpp]: the limit is somewhat of a misnomer -- it's actually an NVRTC specific limitation.  Ideally it wouldn't exist at all.

I'm not sure the best approach to ripping it out, but I don't see why the compiler.cpp file would need to have this logic if it's already checked in the fusion pass

**Review by wanchaol — APPROVED** [review]
It looks like checking producer nodes inputs/outputs should be enough already than checking `flat_inputs/outputs` again in the fusion compiler.

**Comment by yinghai** [comment]
```
./test/test_torch.py:3447:135: B950 line too long (134 > 120 characters)
```

**Comment by wanchaol** [comment]
> ```
> ./test/test_torch.py:3447:135: B950 line too long (134 > 120 characters)
> ```

I think that's a master breakage on linting

**Pull request closed without merging**